ooniapi/services/oonifindings/src/oonifindings/routers/v1.py

```diff
@@ -406,37 +414,38 @@ def update_oonifinding(
 def delete_oonifinding(
     delete_request: OONIFindingWithMail,
     response: Response,
-    authorization: str = Header("authorization"),
     token=Depends(role_required(["admin", "user"])),
-    db=Depends(get_clickhouse_session),
-    settings=Depends(get_settings)
+    db=Depends(get_postgresql_session),
 ):
     """
     Delete an incident
     """
     assert delete_request
-    incident_id = delete_request.incident_id 
-    if token["role"] != "admin":
-        try:
-            verify_user(
-                db,
-                authorization=authorization,
-                jwt_encryption_key=settings.jwt_encryption_key,
-                incident_id=incident_id,
-                email_address=delete_request.email_address,
-                key=settings.account_id_hashing_key,
-            )
-        except:
-            raise 
-        
-    query = "ALTER TABLE incidents DELETE WHERE id = %(incident_id)s"
-    r = raw_query(db, query, {"incident_id": incident_id})
-    optimize_table(db, "incidents")
+    account_id = token["account_id"]
+    incident_id = delete_request.incident_id
+
+    q = db.query(models.OONIFinding).filter(
+        models.OONIFinding.incident_id == incident_id
+    )
+    if token["role"] == "user":
+        q = q.filter(models.OONIFinding.creator_account_id == account_id)
+        if delete_request.email_address != token["email_address"]:
+            raise HTTPException(status_code=403, detail="You are not allowed to delete the incident")
+    else:
+        assert token["role"] == "admin"
+
+    try:
+        q.one()
+    except sa.exc.NoResultFound:
+        raise HTTPException(status_code=404, detail="OONI Finding not found")
+    
+    q.delete()
+    db.commit()
+
     setnocacheresponse(response)
     return {}
 
 
-
 @router.post(
     "/v1/incidents/{action}",
```

Comment:
There is no need for this to be a separate endpoint, it can just be part of the update endpoint where an update request is made with the update column being set.